Source/WebCore/rendering/RenderTableCell.cpp

```diff
@@ -202,7 +203,17 @@ void RenderTableCell::computePreferredLogicalWidths()
     // grids.  We must refresh those grids before the child cells try to use them.
     table()->recalcSectionsIfNeeded();
 
+    // We don't want the preferred width from children to be affected by any
+    // notional height on the cell, such as can happen when a percent sized image
+    // scales up its width to match the available height. Setting a zero override
+    // height prevents this from happening.
+    std::optional<LayoutUnit> contentHeight = this->overridingLogicalHeight();
+    if (contentHeight > -1)
+        setOverridingLogicalHeight(0_lu);
     RenderBlockFlow::computePreferredLogicalWidths();
+    if (contentHeight > -1)
+        setOverridingLogicalHeight(*contentHeight);
+
```

Comment:
```
 LayoutUnit content_height = HasOverrideLogicalContentHeight() ? OverrideLogicalContentHeight() : LayoutUnit(-1);
 if (content_height > -1)
   SetOverrideLogicalContentHeight(LayoutUnit());
 LayoutBlockFlow::ComputePreferredLogicalWidths();
 if (content_height > -1)
   SetOverrideLogicalContentHeight(content_height);

```
Blink seems to use -1 as a magic number, in WebKit this translates to
```
auto overridingLogicalHeight = this->overridingLogicalHeight();
if (overridingLogicalHeight)
  setOverridingLogicalHeight({ });
RenderBlockFlow::computePreferredLogicalWidths();
if (overridingLogicalHeight)
  setOverridingLogicalHeight(*overridingLogicalHeight);

```
though I am not sure if we want a 0 LayoutUnit here or `clearOverridingLogicalHeight`

Reply:
Just to add for history - test fails with `clearOverridingLogicalHeight()`.